oaknational/oak-ai-lesson-assistant
# feat: add users to Hubspot on creation

## Description

We need to send users to hubspot to be able to send them communications and track their email preferences

- Add hubspot client to core package
- Call hubspot client with user details when onboarding

Once this work is deployed I will make sure that existing users are up to date in Hubspot. I already have a script to export preferences to a CSV. If that's not useful for the team, I can adapt it to use this client

## How to test

1. Create a new user with a first and last name. Domain: https://oak-ai-lesson-assistant-j28vnq7f4.vercel-preview.thenational.academy
2. Submit the onboarding form
3. Check [hubspot (staging)](https://app-eu1.hubspot.com/contacts/24947711/objects/0-1/views/all/list)
  1. That the first and last name are there
  2. That the marketing preference is recorded: Go to "Actions" => "View all Properties" => Search for "email_consent_on_ai"

Test with the checkbox both enabled and disabled
Test with an email that already exists in hubspot, and a new account

## Checklist
